tests/unit/coordinator/plugins/test_zookeeper.py

```diff
@@ -92,3 +94,44 @@ async def test_fake_zookeeper_client_get_children_watch() -> None:
         async with client.connect() as connection_2:
             await connection_2.create("/group/of/key_2", b"content")
     assert change_watch.has_changed
+
+
+@pytest.mark.asyncio
+async def test_fake_zookeeper_transaction():
+    client = FakeZooKeeperClient()
+    async with client.connect() as connection:
+        transaction = connection.transaction()
+        transaction.create("/path/to/key_1", b"content")
+        transaction.create("/path/to/key_2", b"content")
+        await transaction.commit()
+        assert await connection.get("/path/to/key_1") == b"content"
+        assert await connection.get("/path/to/key_2") == b"content"
+
+
+@pytest.mark.asyncio
+async def test_fake_zookeeper_transaction_is_atomic():
+    client = FakeZooKeeperClient()
+    async with client.connect() as connection:
+        await connection.create("/path/to/key_2", b"old content")
```

Comment:
The raw connection is just there to prepare the testing situation, the thing under test is only the "second" transaction that needs pre-existing data to be able to fail.

I vaguely prefer not using a transaction for the preparation, this way the test has less opportunities to fail for other reasons than the main topic of the test.

About serializable etc. Zookeeper transaction are just atomic bundles of operations, but they are entirely sent to the server at the "commit" step, there is no question of partial visibility for other clients.
It's exposed as many function calls because the alternative would probably be a large and weakly typed array of operations and not very elegant (in a way, on the client side, a zookeeper transaction is just a builder pattern).